File: Problem125_ValidPalindrome.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        string temp;
        for (int i=0;i<s.length();i++) {
            if (isdigit(s[i]) || isalpha(s[i])) {
                temp += s[i];
            }
        }
        transform(temp.begin(), temp.end(), temp.begin(), ::tolower);
        int l=0, r=temp.length()-1;
        while (l <= r) {
            cout << temp[l] << " " << temp[r] << endl;
            if (temp[l] != temp[r]) return false;
            l+=1;
            r-=1;
        }
        return true;
    }
};

int main() {
    Solution solve;
    string s = "A man, a plan, a canal: Panama";
    bool isPalind = solve.isPalindrome(s);
    cout << isPalind;
}

File: Problem125_ValidPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l = 0
        r = len(s)-1
        while l<=r:
            if not s[l].isdigit() and not s[l].isalpha():
                l+=1
                continue
            if not s[r].isdigit() and not s[r].isalpha():
                r-=1
                continue
            c1 = s[l].lower()
            c2 = s[r].lower()
            print(c1, c2)
            if c1 != c2: return False
            l+=1
            r-=1
        return True

s = "0P"
solve = Solution()
print(solve.isPalindrome(s))